src/test/java/libai/nn/supervised/SVMTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2017 Federico Vera <https://github.com/dktcoding>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package libai.nn.supervised;
+
+import demos.common.SimpleProgressDisplay;
+import java.io.File;
+import java.util.Random;
+import javax.swing.JProgressBar;
+import libai.common.Matrix;
+import libai.common.MatrixIOTest;
+import libai.common.kernels.LinearKernel;
+import libai.nn.NeuralNetwork;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.*;
+
+/**
+ *
+ * @author Federico Vera {@literal <fedevera at unc.edu.ar>}
+ */
+public class SVMTest {
+	@Test
+	public void testDemo() {
+		int n = 100;
+		int t = 40;
+
+		Matrix[] patterns = new Matrix[n + t];
+		Matrix[] ans = new Matrix[n + t];
+
+		Random r = new Random();
```

Comment:
PS: The `new Random()` in `testSaveOpen()` doesn't matter, because the idea is to check that the same Network is read, and convergence is totally unimportant.